### ISSUE TITLE
Deployment of Custom Segmentation Models and Fine-tune Available Segmentation Models

### ISSUE THREAD

**Issue by ali4864** [google-ai-edge/mediapipe#6325]
Is it possible to train custom image segmentation models and deploy on MediaPipe web? As I know we can use some .tflite models on MediaPipe, but it is a question that what are the constrains? What ops are valid to use with WebGL backend? What kind of architecture do it support?

And also is it possible to find-tune available models, like Selfie-Segmentation and Selfie-Segmentation-Multi-Class for web?

**Comment by HarishPermal** [comment]
hi @ali4864 

Thanks for the questions,  yes you can deploy custom TFLite image segmentation models on the web, make sure the exported `.tflite` model follows the [image segmenter metadata writer](https://developers.google.com/edge/api/mediapipe/python/mp/tasks/vision/ImageSegmenter).

Coming to valid ops, you can refer the [official LiteRT / TensorFlow Lite GPU delegate guide](https://developers.google.com/edge/litert/performance/gpu) and there is also a [recommendation](https://developers.google.com/edge/litert/conversion/tensorflow/ops_compatibility) to avoid custom ops and unsupported layers to prevent severe CPU/GPU transfer bottlenecks 

And for input dimensions and model architectures for real-time vision tasks on the web are detailed in the official MediaPipe Solutions web [guide](https://ai.google.dev/edge/mediapipe/solutions/vision/image_segmenter)

For fine-tuning you can use TensorFlow using public segmentation datasets (such as COCO-Stuff, ADE20K, or custom masks) convert it to TFLite via `tf.lite.TFLiteConverter` [Ref-1](https://developers.google.com/edge/litert/conversion/tensorflow/overview),  

hope this helps

**Comment by ali4864** [comment]
Thank @HarishPermal very much for your clarification. I managed to deploy a custom segmentation model successfully for use on Mediapipe web (CPU backend), but we need to be aware that for WebGL backend it is a little complicated because of limited support for some ops. Regarding the second problem (fine-tuning available segmentation models), I need to clarify that I mean fine-tuning available segmentation models of Mediapipe, for example Mediapipe Multiclass Segmentation, I think it is impossible because we don't have access to its Tensorflow or Keras models! I think for similar tasks we need to design and train our own models, and as Mediapipe's models are very optimized, it will be very helpful to learn their models' architectures, are there any documentation about details of your model architectures? For example we only know Mediapipe Multiclass Segmentation is based on vision transformers, not other information!
